Products/CMFCore/FSFile.py

```diff
@@ -76,7 +76,8 @@ def _get_content_type(self, file, body, id, content_type=None):
             # Last resort: Use the (imperfect) content type guessing
             # mechanism from OFS.Image, which ultimately uses the
             # Python mimetypes module.
-            if not isinstance(body, six.string_types):
+            if not isinstance(body, six.string_types) and \
+                    getattr(body, 'data', None) is not None:
```

Comment:
This needs a test too. Even the commit message does not tell why which needs to be changed.

Reply:
There is an alternative fix in ce515fb with the same result on the tests: It reduces the number of errors in the Python 3.6 test run by 10.

Reply:
ce515fb was merged to master after approval of @dataflake, so I'd suggest to use it here, too.